src/lib/mina_networking/mina_networking.ml

```diff
@@ -1438,6 +1438,34 @@ let create (config : Config.t)
   let%map gossip_net =
     Gossip_net.Any.create config.creatable_gossip_net rpc_handlers
   in
+  (* Initialize the ban loop. *)
+  don't_wait_for
+    (Strict_pipe.Reader.iter (Trust_system.ban_pipe config.trust_system)
+       ~f:(fun (banned_peer, expiration) ->
+         don't_wait_for
```

Comment:
Doesn't the order of execution here mean that the ban may be effected before the notification is sent, depending on the scheduler behaviour? Especially since the RPC handshake needs a round-trip to the helper, whereas the banlist update is one-shot, I would expect the ban_notify to pretty much never be received.

Reply:
That's a good point, I didn't consider that. Thanks!